install-deps.sh

```diff
@@ -167,6 +167,8 @@ function activate_virtualenv() {
     . $env_dir/bin/activate
 }
 
+test "$1" = "skip-virtualenv" && exit 0
```

Comment:
Tox is nice for virtualenv management, but the tests could run outside of tox during "make check".

Using system packages during "make check" would work without network access in the scenario you describe. It would be simpler because the process would not have to rely on PyPI's availability at all.